### PR TITLE
tls-endpoint: allow an ip to listen on being specified on command line

### PR DESCRIPTION
**Pull request by hannesm**
for fixing the other part of #143 (or the suggestion in it)

### CODE REVIEW

File: src/vmm_lwt.ml [hunk 1]
```diff
@@ -12,13 +12,13 @@ let safe_close fd =
...
 
-let port_socket port =
+let port_socket ip port =
   let open Lwt_unix in
   let s = socket PF_INET6 SOCK_STREAM 0 in
   set_close_on_exec s ;
   setsockopt s SO_REUSEADDR true ;
   setsockopt s IPV6_ONLY false ;
-  bind s (ADDR_INET (Unix.inet6_addr_any, port)) >>= fun () ->
+  bind s (ADDR_INET (Ipaddr_unix.to_inet_addr ip, port)) >>= fun () ->
```
Comment (reynir) [inline src/vmm_lwt.ml:21]: Hmm, I worry if we specify `0.0.0.0` that we get the same error on Linux.

Comment (reynir) [reply src/vmm_lwt.ml:21]: Yes, we do.

Comment (hannesm) [reply src/vmm_lwt.ml:21]: uhm, so should we change some code? we could use above `PF_INET` if we have an ipv4 address?

Comment (reynir) [inline src/vmm_lwt.ml:21]:
```suggestion
  let ip = Ipaddr.V6 (Ipaddr.to_v6 ip) in
  bind s (ADDR_INET (Ipaddr_unix.to_inet_addr ip, port)) >>= fun () ->
```

Comment (reynir) [reply src/vmm_lwt.ml:22]: I think we can do a dance by converting to ipv6 with `Ipaddr.(V6 (to_v6 ip))` as I just suggested below.

Comment (hannesm) [reply src/vmm_lwt.ml:22]: 👍🏾

Comment (reynir) [reply src/vmm_lwt.ml:22]: I just tested binding using the above code with "0.0.0.0", and it doesn't fail, but I don't see it listening in the output from `netstat -tlnp`. I will try to test some more...

Comment (reynir) [reply src/vmm_lwt.ml:22]: This article suggests that there may be issues with using ::ffff:0.0.0.0 on some platforms: https://lore.kernel.org/all/87tz5s1pxl.fsf@nemi.mork.no/T/

**Comment by hannesm** [comment]
So, what is the opinion on this PR? We could:
- just close it (until someone requests such a feature)
- in the bind have code paths depending on the ip protocol version provided
- just remove albatross_tls_endpoint (and advertise the inetd service more)
- select a single ip protocol version to use

What do you think @reynir? I'm fine with option 2 since it seems to solve the remaining issues.

**Comment by hannesm** [comment]
I propose [a307e58](https://github.com/roburio/albatross/pull/145/commits/a307e588f6a08a9ef6ba5733d3c20785a83b91f7) (part of this PR), and think this can be merged and shouldn't exhibit strange behaviour.

**Review by reynir — APPROVED** [review]

**Pull request merged**